app/controllers/api/assignments_controller.rb

```diff
@@ -0,0 +1,133 @@
+            render json: @assignment, status: :ok
+        else
+            render json: { error: 'Could not update action item' }, status: :unprocessable_entity
+        end
+    end
+
+    def destroy
+        authorize @assignment
+        action_item = @assignment.action_item
+        if @assignment.destroy
+            if action_item.assignments.empty?
+                action_item.destroy
+            end
+            render json: {}, status: :ok
+        else
+            render json: { error: 'Failed to delete action item' }, status: :unprocessable_entity
+        end
+    end
+
+    def create_template
+        @template = authorize ActionItem.new(action_item_params), :create?
+        @template[:is_template] = true
+
+        if @template.save
+            render json: @template, status: :created
+        else
+            render json: { error: 'Could not create template' }, status: :unprocessable_entity
+        end
+    end
+
+    def show_template
+        authorize @template, :show?
+        render json: @template, status: :ok
+    end
+
+    def update_template
+        authorize @template, :update?
+        if @template.update(action_item_params)
+            render json: @template, status: :ok
+        else
+            render json: { error: 'Could not update template' }, status: :unprocessable_entity
+        end
+    end
+
+    def destroy_template
+        authorize @template, :destroy?
+        if @template.destroy
+            render json: @template, status: :ok
+        else
+            render json: { error: 'Failed to delete action item' }, status: :unprocessable_entity
+        end
+    end
+
+    def user_not_authorized
+        render json: { error: 'You are not authorized to perform this action' }, status: :unauthorized
+    end  
+
+    private
+    
+    def set_template
+        @template = ActionItem.find(params[:id])
+    rescue ActiveRecord::RecordNotFound
+        render json: { error: 'Could not find Action Item Template' }, status: :not_found
+    end
+
+    def set_assignment
+        @assignment = Assignment.find(params[:id])
+    rescue ActiveRecord::RecordNotFound
+        render json: { error: 'Could not find Action Item' }, status: :not_found
+    end
+
+    def prepare_bulk_assignment(assigned_to_ids, action_item_id)
+        bulk_assignment_params = []
+        single_assignment_params = assignment_params
+        assigned_to_ids.each do |id|
+            bulk_assignment_params.append(single_assignment_params.merge(assigned_to_id: id, 
+                                                                         action_item_id: action_item_id))
+        end
+        return bulk_assignment_params
+    end
+
+    def action_item_params
+        action_item_param = params.require(:assignment).permit(:title,
+                                                                :description)
+    end
+
+    def assigned_to_ids
```

Comment:
where is this function used?

Reply:
Line 12 in `prepare_bulk_assignment(assigned_to_ids, @action_item.id).each do ...`